### PR TITLE
Add flags for disabling the diagnostic server or setting its port

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/7557

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth

File: shell/common/shell.cc [hunk 1]
```diff
@@ -66,8 +78,8 @@ base::LazyInstance<NonDiscardableMemoryAllocator> g_discardable;
 void ServiceIsolateHook(bool running_precompiled) {
   if (!running_precompiled) {
     const blink::Settings& settings = blink::Settings::Get();
-    if (settings.enable_observatory)
-      DiagnosticServer::Start();
+    if (settings.enable_observatory && settings.enable_diagnostic)
```
Comment (Hixie) [inline shell/common/shell.cc]: why do you require `settings.enable_observatory` to be on as well here?

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
LGTM modulo my nit

File: shell/common/shell.cc (hunk 1 above)
Comment (jason-simmons) [inline shell/common/shell.cc]: This was preexisting behavior - I think the assumption was that anyone who disabled the observatory wouldn't want the diagnostic server either.

Comment (jason-simmons) [inline shell/common/shell.cc]: Changed this to just check for enable_diagnostic

**Pull request merged**